libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/http/BatchingHttpInterceptor.kt

```diff
@@ -111,6 +99,11 @@ class BatchingHttpInterceptor @JvmOverloads constructor(
     }
     if (sendNow) {
       executePendingRequests()
+    } else {
+      scope.launch {
+        delay(batchIntervalMillis)
+        executePendingRequests()
+      }
```

Comment:
This changes the behaviour. For a bunch of grouped requests, the latency is now always `batchIntervalMillis` while before it was on average `batchIntervalMillis/2`.

Of course it might be better for other patterns of requests I believe grouped requests is the typical use case, send n requests in a very short timeframe when opening a screen and then get all of them all at once. So not sure we want to change this?

Reply:
Sorry for the lack of context! I changed this because in the current implementation, we would have a never-ending loop (the `while (true)` above).  I agree that it changes the behavior though. Maybe we can keep the loop but suspend until there's at least one pending request?

Reply:
> Maybe we can keep the loop but suspend until there's at least one pending request?

Yup, that'd work!